src/Tests/DocumentBundleTest.php

```diff
@@ -0,0 +1,315 @@
+<?php
+
+namespace Drupal\media\Tests;
+
+use Drupal\simpletest\WebTestBase;
+
+/**
+ * Ensures that media bundle for document can be created.
+ *
+ * @group media
+ */
+class DocumentBundleTest extends WebTestBase {
+  /**
+   * Exempt from strict schema checking.
+   *
+   * @see \Drupal\Core\Config\Testing\ConfigSchemaChecker
+   *
+   * @var bool
+   */
+  protected $strictConfigSchema = FALSE;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'media',
+    'media_entity',
+    'media_entity_document',
+    'node',
+  ];
+
+  /**
+   * The test media bundle.
+   *
+   * @var \Drupal\media_entity\MediaBundleInterface
+   */
+  protected $testBundle;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->testBundle = $this->container->get('entity_type.manager')->getStorage('media_bundle')->load('document');
+
+    $adminUser = $this->drupalCreateUser([
+      'view media',
+      'create media',
+      'update media',
+      'update any media',
+      'delete media',
+      'delete any media',
+      'access media overview',
+    ]);
+    $this->drupalLogin($adminUser);
+  }
+
+  /**
+   * Tests document media bundle creation from config files.
+   */
+  public function testMediaBundleCreationFromModule() {
+    $type_configuration = [
+      'source_field' => 'field_document',
+    ];
+    $field_map = [
+      'mime' => 'field_mime_type',
+      'size' => 'field_document_size',
+    ];
+
+    $this->assertTrue((bool) $this->testBundle, 'The media bundle from default configuration has been created in the database.');
+    $this->assertEqual($this->testBundle->get('label'), 'Document', 'Correct label detected.');
+    $this->assertEqual($this->testBundle->get('description'), 'Use Document for uploading document files such as PDF.', 'Correct description detected.');
+    $this->assertEqual($this->testBundle->get('type'), 'document', 'Correct plugin ID detected.');
+    $this->assertEqual($this->testBundle->get('type_configuration'), $type_configuration, 'Type configuration correct.');
+    $this->assertEqual($this->testBundle->get('field_map'), $field_map, 'Correct field map detected.');
+  }
+
+  /**
+   * Tests thumbnails of the document items.
+   */
+  public function testDocumentItemThumbnail() {
+    // Test thumbnail of PDF file.
+    $file = drupal_get_path('module', 'media') . '/files/Test.pdf';
```

Comment:
I think that this test could be simplified if it would be converted to a loop (it seems that we are doing more or less same things for each file).